boa/src/builtins/array/mod.rs

```diff
@@ -48,16 +48,13 @@ impl Array {
                 .expect("Could not get global object"),
         ));
         array.set_data(ObjectData::Array);
-        array.borrow().set_internal_slot(
-            INSTANCE_PROTOTYPE,
-            interpreter
-                .realm()
-                .environment
-                .get_binding_value("Array")
-                .expect("Array was not initialized")
-                .borrow()
-                .get_field(PROTOTYPE),
-        );
+        array.as_object_mut().unwrap().prototype = interpreter
```

Comment:
I would use `expect()` here, saying something like "an array was not an object" or something like that.